### ISSUE TITLE
Request: Allow for uploading new versions of files

### ISSUE THREAD

**Issue by mrbiber** [PeerioTechnologies/peerio-client#22]
If I share a file with someone, I should be able to update that file so that the update propagates to all people I've shared it with. This would make cooperation much easier, as you would then not send dozens of files with the same name.

**Comment by hdf** [comment]
You can delete previously sent files from the in-boxes of everyone that received it even if it was just passed along, and did not come from you directly. Resend the new one, and you are there. This feature basically already exists.

**Comment by mrbiber** [comment]
Yes, I see that one could do it this way. But this seems unintuitive and complicated. Most people might expect a way to directly update a file. Also, later versioning functionality could be added (so that you could still access the old versions of a file). If you delete and resend, the information that this is an update to a specific file is lost.

**Comment by hdf** [comment]
Why not send the file with the version in the file name? There is also search functionality in Peerio, so it's no problem, if the different versions don't sit right under each other.

**Comment by mrbiber** [comment]
Of course, there are workarounds for this use-case. But filename-versioning clearly is a workaround that's complicated to use. If people want this feature, it would be more convenient to have it properly implemented. That other file exchange platforms (Google Drive, SpiderOak) support file versioning might be evidence that this is a useful feature to have.

In any case, my request was adressed to the developers. They should evaluate and perhaps research if this adds value for the users. Users who do not need this feature just can refrain from using it. I suggest we leave it for the developers to say whether this fits into their vision of the software.

**Comment by kaepora** [comment]
This is an interesting feature, but not something I'd consider high-priority. Will leave this open for future looking-into.
